apps/api/src/middleware/active-comp-filter.middleware.ts

```diff
@@ -0,0 +1,71 @@
+import { eq } from "drizzle-orm";
+import { NextFunction, Request, Response } from "express";
+
+import { config } from "@/config/index.js";
+import { db } from "@/database/db.js";
+import { competitions } from "@/database/schema/core/defs.js";
+import { middlewareLogger } from "@/lib/logger.js";
+import { ApiError } from "@/middleware/errorHandler.js";
+import { COMPETITION_STATUS } from "@/types/index.js";
+
+/**
+ * Active Competition Filter Middleware
+ *
+ * This middleware checks if there is an active competition. If no active competition
+ * is found, it responds with a 403 Forbidden error.
+ */
+export const activeCompMiddleware = function () {
+  return async function (req: Request, res: Response, next: NextFunction) {
+    try {
+      const activeCompetition = await getActiveComp();
+      if (!activeCompetition) {
+        middlewareLogger.debug(
+          "Active comp middleware: No active competition found",
+        );
+        throw new ApiError(403, "No active competition");
```

Comment:
Simply throw a 403 if there isn't an active competition.  This middleware can be used by any endpoint that is only valid when a competition is happening.